### PR TITLE
Return ETIMEDOUT when TCP request times out

### PR DESCRIPTION
**Pull request by calthomson**
Add timeout handling for TCP requests.

## Test Plan
How I tested this locally:

* Pull this branch locally.
* In a project that uses this lib (ex. `node-counters-client`) run `npm link ../jsonrpc2.js` (or whatever relative path you have in your local env)
* Inside gateway-api run `npm link ../node-counters-client` (or whatever relative path you have to this project)
* Run gateway-api locally against stage: `robo sshuttle; treb local -e stage .run/gateway-api.yml -- make server`
* Run the following GraphQL query to hit the `counters-service`:

```
query app__getHomeData($workspaceSlug: Slug!) {
  workspace: workspace(slug: $workspaceSlug) {
    id
    slug
    sources {
      legacySchemaTracks
    }
  }
}
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,6 +79,14 @@ class Client {
     const socket = net.connect(this.port, this.hostname)
     socket.setTimeout(options.timeout || this.timeout)
 
+    socket.on('timeout', () => {
+      socket.end()
+      const err = new Error('Request timed out')
```
Comment (Rowno) [inline index.js]: Might be nice to include the `hostname` in the error message so it's easy to tell what service the error is coming from.

File: index.js [hunk 2]
```diff
@@ -79,6 +79,14 @@ class Client {
...
+      err.code = 'ETIMEDOUT'
+
+      fn(err)
+    })
+
     socket
       .on('error', fn)
```
Comment (colinking) [inline index.js]: non-blocking: Should we move the `on('timeout')` to here, with the other handlers?

Comment (calthomson) [reply index.js]: Done!

File: index.js [hunk 3]
```diff
@@ -80,6 +80,13 @@ class Client {
     socket.setTimeout(options.timeout || this.timeout)
 
     socket
+      .on('timeout', () => {
+        socket.end()
+        const err = new Error('Request timed out. Hostname: ', this.hostname)
```
Comment (Rowno) [inline index.js]: I don't think this will work, I'm pretty sure `Error` only takes one parameter.

Comment (Rowno) [reply index.js]: Looks like the second parameter is supposed to be used for the filename?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

Comment (calthomson) [reply index.js]: You're right, whoops, fixing it here: https://github.com/segmentio/jsonrpc2.js/pull/27/files

**Pull request merged**